improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py

```diff
@@ -28,62 +28,725 @@
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                    ],
+                    [
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 0.4, 1.0, 1.0, 1.0],
+                        [0.4, 0.4, 0.4, 1.0, 1.0],
+                        [1.0, 0.4, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                    ],
+                ],
+                [
+                    [
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                    ],
+                    [
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                    ],
+                ],
+                [
+                    [
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                    ],
+                    [
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                        [1.0, 1.0, 1.0, 1.0, 1.0],
+                    ],
+                ],
+            ]
+        )
+        percentiles = np.array([10, 50, 90])
+        radius = 2000.0
+        result = GeneratePercentilesFromANeighbourhood(
+            radius, percentiles=percentiles
+        ).process(cube)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_single_point_lat_long(self):
+        """Test behaviour for a single grid cell on lat long grid."""
+
+        data = np.ones((16, 16), dtype=np.float32)
+        data[7, 7] = 0
+        cube = set_up_variable_cube(data, spatial_grid="latlon",)
+
+        msg = "Unable to convert from"
+        radius = 6000.0
+        with self.assertRaisesRegex(ValueError, msg):
+            GeneratePercentilesFromANeighbourhood(radius).process(cube)
+
+    def test_single_point_masked_to_null(self):
+        """Test behaviour with a masked non-zero point.
+        The underlying numpy.percentile base behaviour does not support
+        masked arrays so raise an error.
+        """
+        mask = np.zeros_like(self.cube.data)
+        mask[2, 2] = 1
+        self.cube.data = np.ma.masked_array(self.cube.data, mask=mask)
+        percentiles = np.array([10, 50, 90])
+        radius = 2000.0
+        message = "The use of masked data is not yet implemented."
```

Comment:
As noted on the PR, this message requires updating.

Reply:
Done